### PR TITLE
fix!: allow ctx.edit to edit component responses when not deferred

### PR DESCRIPTION
**Pull request by sillyElvy**
This PR allows messages to be edited as a component response using ctx.edit
It also prevents message contents from being removed when editing the message
## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [x] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent): #435 
- [ ] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [ ] New feature/enhancement
  - [x] Bugfix

### CODE REVIEW

File: interactions/context.py [hunk 1]
```diff
@@ -182,15 +182,24 @@ async def send(
         :return: The sent message as an object.
         :rtype: Message
         """
+        if content is None and self.message:
+            _content = self.message.content
```
Comment (i0bs) [inline interactions/context.py:190]: Why do you have a separate conditional here on top of the ternary operation?

Comment (sillyElvy) [reply interactions/context.py:190]: Because (iirc) I made it so that can use `send` when you defer components to edit the original message

Comment (i0bs) [reply interactions/context.py:190]: `_content` will be overwritten using this logic. Assuming that `message` is an attribute with a value, the second declaration of the variable in this method will simply reset the value of it. The ternary fails to consider the latter here, turning this into a lossy solution.

**Comment by i0bs** [comment]
Due to write access inheritance on forks of this project as the owner/maintainer, I was able to make commits fixing this. I will be merging this now.

**Pull request merged**